### ISSUE TITLE
[UI] - Folder upload not possible in some cases

### ISSUE THREAD

**Issue by mmartin24** [epinio/ui#130]
**Describe the bug**
I am unable to upload `sample-app` folder. See video:
[uploadSampleAppNotAllowed.zip](https://github.com/epinio/epinio/files/8851238/uploadSampleAppNotAllowed.zip)
![image](https://user-images.githubusercontent.com/37271841/172327205-e7d15777-6b7c-405d-9a1d-2c4f312aa2fb.png)

However it seems possible to be able to upload some other types of folders like the one in the video (however the same error is triggered)
[uploadFolderAllowed.zip](https://github.com/epinio/epinio/files/8851241/uploadFolderAllowed.zip)
![image](https://user-images.githubusercontent.com/37271841/172327522-5fee3d56-8b7f-4a50-96a6-91bcfee2eb2d.png)

Hint: Make sure to search through the existing issues, as they often contain important information on recent changes.

**To Reproduce**
Steps to reproduce the behavior, e.g.:

1. Install Standalone Epinio UI
2. Attempt to upload `sample-app` from UI

**Expected behavior**
App should be uploaded and no errors should appear on the background

**Logs**
If applicable, add logs to help explain your problem.
```
vue.runtime.esm.js:1897 TypeError: Invalid attempt to iterate non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.
```

**Comment by richard-cox** [comment]
@mmartin24 @jimmykarily Can you confirm the title of the browser dialog window that pops up when you click `select folder`?

**Comment by jimmykarily** [comment]
The error I get:

```
Invalid attempt to iterate non-iterable instance. In order to be iterable, non-array objects must have a [Symbol.iterator]() method. 
```

console error points here:

https://github.com/rancher/dashboard/blob/662544ec8d1028fa9af961bf02028e21db10cc57/pkg/epinio/components/application/AppSource.vue#L163

**Comment by jimmykarily** [comment]
So we now know what the problem is. If the folder has only one file, it's not passing an array to the `onFolderSelected`. If I just create an empty file next to `index.html` it then works.

**Comment by mmartin24** [comment]
> 
@richard-cox , In my case is "Open File"
![image](https://user-images.githubusercontent.com/37271841/172639358-3c554e70-6205-45bd-8f56-fed61df13386.png)

**Comment by richard-cox** [comment]
Thanks both, i've pushed a fix to `epinio-dev` and will be available to validate in the next standalone release

**Comment by juadk** [comment]
### Fixed in the latest UI build (`v0.9.0-0.0.3)` :heavy_check_mark: 
I uploaded the sample-app folder (only one file in) and deployed the app successfully without any error in the web dev console.
